Negotiation/FormatNegotiator.php

```diff
@@ -0,0 +1,113 @@
+ */
+
+namespace FOS\RestBundle\Negotiation;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+use Negotiation\Negotiator as BaseNegotiator;
+use Negotiation\Accept;
+
+class FormatNegotiator extends BaseNegotiator
+{
+    /**
+     * @var array
+     */
+    private $map = array();
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * Constructor.
+     */
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * @param RequestMatcherInterface $requestMatcher
+     * @param array                   $options
+     */
+    public function add(RequestMatcherInterface $requestMatcher, array $options = array())
+    {
+        $this->map[] = array($requestMatcher, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     * The best format is also determined in function of the bundle configuration.
+     */
+    public function getBest($header, array $priorities = array())
+    {
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (empty($header)) {
+            $header = $request->headers->get('Accept');
+        }
+
+        foreach ($this->map as $elements) {
+            // Check if the current RequestMatcherInterface matches the current request
+            if (null === $elements[0] || !$elements[0]->matches($request)) {
+                continue;
+            }
+
+            $options = &$elements[1]; // Do not reallow memory for this variable
+
+            if (!empty($options['stop'])) {
+                throw new StopFormatListenerException('Stopped format listener');
+            }
+            if (empty($options['priorities']) && empty($priorities)) {
+                if (!empty($options['fallback_format'])) {
+                    return new Accept($request->getMimeType($options['fallback_format']));
+                }
+                continue;
+            }
+            if ($options['prefer_extension']) {
+                // ensure we only need to compute $extensionHeader once
+                if (!isset($extensionHeader)) {
+                    if (preg_match('/.*\.([a-z0-9]+)$/', $request->getPathInfo(), $matches)) {
+                        $extension = $matches[1];
+                    }
+
+                    // $extensionHeader will now be either a non empty string or an empty string
+                    $extensionHeader = isset($extension) ? (string) $request->getMimeType($extension) : '';
+                    if ($header && $extensionHeader) {
+                        $extensionHeader = ','.$extensionHeader;
+                    }
+                }
+                if ($extensionHeader) {
+                    $header .= $extensionHeader.'; q='.$options['prefer_extension'];
+                }
+            }
+
+            $mimeTypes = $this->normalizePriorities(empty($priorities) ? $options['priorities'] : $priorities);
```

Comment:
Hum, the solution may be to copy the ``Request::$formats``array... Or as you said, we can also force the users to only provide mime types.